srml/executive/src/lib.rs

```diff
@@ -609,15 +621,8 @@ mod tests {
 				));
 
 				if lock == WithdrawReasons::except(WithdrawReason::TransactionPayment) {
-					assert_eq!(
-						Executive::apply_extrinsic(xt).unwrap(),
-						Err(DispatchError {
-							module: None,
-							error: 0,
-							message: Some("account liquidity restrictions prevent withdrawal"),
-						}),
-					);
-					// but tx fee has been deducted. the transaction failed on transfer, not on fee.
+					assert!(Executive::apply_extrinsic(xt).is_ok());
```

Comment:
```suggestion
					assert!(Executive::apply_extrinsic(xt).map(Result::is_ok).unwrap_or(false));
```

Before, you checked if the transaction itself returned an error. If you want to know if the transaction could be executed successfully, you need to check that the result is `Ok(Ok(()))`.